File: Task-3_factorial calculation/factorial.py
def factorial(n):
    """
    Recursive function to compute the factorial of a number.
    Handles edge cases for 0 and negative numbers.
    """
    if n < 0:
        raise ValueError("Factorial is not defined for negative numbers.")
    elif n == 0:
        return 1
    else:
        return n * factorial(n - 1)

if __name__ == "__main__":
    try:
        num = int(input("Enter a number to calculate its factorial: "))
        result = factorial(num)
        print(f"Factorial of {num} is {result}")
    except ValueError as e:
        print(e)

File: Task-3_factorial calculation/Factorial.java
import java.util.Scanner;

public class Factorial {
    // Recursive method to calculate factorial
    public static long factorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("Factorial is not defined for negative numbers.");
        } else if (n == 0) {
            return 1;
        } else {
            return n * factorial(n - 1);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number to calculate its factorial: ");
        try {
            int num = scanner.nextInt();
            long result = factorial(num);
            System.out.println("Factorial of " + num + " is " + result);
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        } catch (Exception e) {
            System.out.println("Invalid input. Please enter an integer.");
        }
        scanner.close();
    }
}
